[NETBEANS-54] Module Review websvc.saas.codegen.java

tomazst opened the pull request:
Apache license headers added. Closes #211

matthiasblaesing committed:
[NETBEANS-54] Module review xml.text.obsolete90

- no external library
- checked Rat report; added license header to testfiles and ensured 
  tests still run cleanly, removed unused file, that only differs from
  neighbouring file by license
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.core

- no external library
- checked Rat report; added license header to testfiles
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.retriever

- no external library
- checked Rat report; added license header to testfiles
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.multiview

- no external library
- checked Rat report; added license header to testfiles, manually 
  updated unconverted license headers
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xsl

- no external library
- checked Rat report; added testfiles to exclude list for rat
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.text

- no external library
- checked Rat report; added license to many test files and adjusted
  tests to match these; exclude GUI visible sample data and special
  test data
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.axi

- no external library
- checked Rat report: excluded test-data from rat (no IP value) 
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.xam

- no external library
- checked Rat report: manually updated missed license header and added
  missing headers
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.schema.completion
    
- no external library
- checked Rat report: excluded test-data from rat (no IP value), updated
  XMLSchema.xsd and added w3c document and software license and
  made sure license is picked up by rat report
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.tax
    
- no external library
- checked Rat report: added missing headers, ignored i10n.list
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.jaxb.api

- external library jsr173_api.jar was removed, as StAX api was moved
  into java platform
- external libraries jaxb-api and activation were moved to maven
  coordinates and version was corrected
- checked Rat report: updates license header in Bundle.properties
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.xdm

- no external library
- checked Rat report: excluded test-data from rat (no IP value)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.catalog

- removed external binary (contained XML Schemas and DTDs), added 
  license header to files and added them to the source code, where the
  binary was expanded at build time
- checked Rat report: added missing headers
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.lexer

- no external library
- checked Rat report: excluded test-data from rat (no IP value)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.tools

- no external library
- checked Rat report: excluded test-data from rat (no IP value)
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.schema.model

- no external library
- checked Rat report: excluded test-data from rat (no IP value), covered
  zip files contains simple XSD samples as groups
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.wsdl.model

- no external library
- checked Rat report: manully updated missed license headers
- readded undonated wsdl-2004-08-24.xsd from source and added license
  to it (fixes functionality)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

tomazst committed:
[NETBEANS-54] Module Review websvc.saas.codegen.java

Apache license headers added.

matthiasblaesing committed:
[NETBEANS-54] Modules review xml

- no external library
- checked Rat report: excluded test-data from rat (no IP value),
  updated/added license headers, exclude templates from rat (jira
  issue was raised)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

emilianbold commented on websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/resources/signedurl.properties:
```diff
@@ -1,3 +1,23 @@
+#
+# The format should handle comments
+# And also empty lines
+
```
Why are these added:

>\#  The format should handle comments
>\# And also empty lines

tomazst committed:
[NETBEANS-54] Module Review websvc.saas.codegen.java

Changed license headers.

matthiasblaesing committed:
Add support for specifying classifier and extension for maven binaries

Some modules not only load plain artifacts from maven, but also need
additional parts. In maven the primary artifact can be accompanied by
additional files, that have the same base name, but are seperated by
a classifier (javadoc and sources for example).

What is more it is possible to publish artifacts with different
extensions than "jar".

This commit introduces support for this. The format for maven
coordinates is expanded from:

group:artifact:version

to

group:artifact:version:classifier@extension

Both classifier and extension are optional and the default for extension
is "jar", so this change is compatible.

The format follows the short version of gradle dependencies:

https://docs.gradle.org/4.2.1/dsl/org.gradle.api.artifacts.dsl.DependencyHandler.html

Section "External dependencies"

Commit:
[NETBEANS-54] Module Review libs.asm

Commit:
[NETBEANS-54] Module Review libs.junit4

juneau001 committed:
[NETBEANS-54] Module Review j2ee.metadata.model.support

tomazst committed:
[NETBEANS-54] Module Review testng

trevlewis committed:
[NETBEANS-54] Module Review testng.ui

* Updated license in Bundle.properties
* Add exclude for testng.ui template files

matthiasblaesing committed:
Merge branch 'maven_classifier_extension'

trevlewis committed:
[NETBEANS-54] Module Review versioning

matthiasblaesing committed:
Merge branch 'pr/103'

matthiasblaesing committed:
Merge branch 'master' of https://git-wip-us.apache.org/repos/asf/incubator-netbeans

matthiasblaesing committed:
Merge branch 'pr/102'

matthiasblaesing committed:
Merge branch 'pr/110'

vieiro committed:
[NETBEANS-54] Module Review libs.lucene

  - Added lucene notice file from https://github.com/apache/lucene-solr/blob/releases/lucene-solr/3.5.0/lucene/NOTICE.txt
  - Added the maven coordinates.
  - No other issues found.

bsungur committed:
[NETBEANS-54] Module Review java.lexer

Add Rat exclude for test data.

trevlewis committed:
[NETBEANS-54] Module Review versioning.util

Exclude test data.

trevlewis committed:
[NETBEANS-54] Module Review websvc.saas.kit

Exclude release/VERSION.txt

38leinaD committed:
[NETBEANS-54] Module Review subversion

batouxclass committed:
[NETBEANS-54] Module Review spring.beans

Replaced the old license with the Apache license for spring.beans
module.

Excluded the template file from Rat

trevlewis committed:
[NETBEANS-54] Module Review testng.ant

bjansen committed:
[NETBEANS-54] Module Review extbrowser

tuyenngoctran committed:
[NETBEANS-54] Module Review java.sourceui

tuyenngoctran committed:
[NETBEANS-54] Module Review javadoc

Update license in AddTagFixTest.java.
Exclude test file index-4.title from Rat.

tomazst committed:
[NETBEANS-54] Module Review projectuiapi

Excluded files with test data from Rat.

matthiasblaesing committed:
Merge branch 'pr/112'

matthiasblaesing committed:
Merge branch 'master' of https://git-wip-us.apache.org/repos/asf/incubator-netbeans

bsungur committed:
[NETBEANS-54] Module Review languages.diff

Exclude DiffExample.diff from Rat since a license header would be visible to users in Fonts and Color settings.

trevlewis committed:
[NETBEANS-54] Module Review selenium2.java

tomazst committed:
[NETBEANS-54] Module Review projectui

Exclude license files from Rat.

trevlewis committed:
[NETBEANS-54] Module Review spellchecker.dictionary_en

tomazst committed:
[NETBEANS-54] Module Review css.prep

Added apache license headers and excludes to build.xml

emilianbold commented on websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/resources/ApiKeyAuthenticator.template:
```diff
@@ -1,3 +1,24 @@
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
```
This .template file should probably be excluded the nbbuild/build.xml "rat" task since it's user visible.

emilianbold commented on websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/resources/HttpBasicAuthenticator.template:
```diff
@@ -1,3 +1,24 @@
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
```
Another template file that should be excluded.

emilianbold commented on websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/resources/RestConnection.template:
```diff
@@ -1,3 +1,24 @@
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
```
Another template.

emilianbold commented on websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/resources/RestResponse.template:
```diff
@@ -1,3 +1,24 @@
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
```
Another template.

emilianbold commented on websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/resources/apikey.properties:
```diff
@@ -1,3 +1,20 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
```
I wonder, since this is part of the `codegen` package, does the user get to see this? Because if so, we should also exclude it. If not, it's OK.

Can you check?

tomazst committed:
Merge remote-tracking branch 'origin/websvc.saas.codegen.java' into websvc.saas.codegen.java

emilianbold reviewed (changes requested):
I see 197 files changed.
